```
mypy.......................................................................Failed
- hook id: mypy
- duration: 214.27s
- exit code: 1

pandas/core/internals/blocks.py:1492: error: Incompatible return value type (got "List[Block]", expected "List[Self]")  [return-value]
pandas/core/internals/blocks.py:1497: error: List item 0 has incompatible type "Block"; expected "Self"  [list-item]
Found 2 errors in 1 file (checked 1418 source files)

mypy (stubtest)............................................................Failed
- hook id: stubtest
- duration: 122.98s
- exit code: 1

error: not checking stubs due to mypy build errors:
pandas/core/internals/blocks.py:1492: error: Incompatible return value type (got "List[Block]", expected "List[Self]")  [return-value]
pandas/core/internals/blocks.py:1497: error: List item 0 has incompatible type "Block"; expected "Self"  [list-item]
```

--- a/pandas/core/internals/blocks.py
+++ b/pandas/core/internals/blocks.py
@@ -1463,7 +1463,7 @@ class Block(PandasObject):
         new_values = algos.diff(self.values.T, n, axis=0).T
         return [self.make_block(values=new_values)]
 
-    def shift(self, periods: int, fill_value: Any = None) -> list[Self]:
+    def shift(self, periods: int, fill_value: Any = None) -> list[Block]:
         """shift the block by periods, possibly upcast"""
         # convert integer to float if necessary. need to do a lot more than
         # that, handle boolean etc also
@@ -1494,7 +1494,7 @@ class Block(PandasObject):
         else:
             values = cast(np.ndarray, self.values)
             new_values = shift(values, periods, axis, casted)
-            return [self.make_block(new_values)]
+            return [self.make_block_same_class(new_values)]
 
     @final
     def quantile(
@@ -1645,7 +1645,7 @@ class EABackedBlock(Block):
 
     values: ExtensionArray
 
-    def shift(self, periods: int, fill_value: Any = None) -> list[Self]:
+    def shift(self, periods: int, fill_value: Any = None) -> list[Block]:
         """
         Shift the block by `periods`.
 

